sympy/core/function.py

```diff
@@ -1806,6 +1806,17 @@ def _eval_is_commutative(self):
         return self.expr.is_commutative
 
     def doit(self):
+        if isinstance(self.expr,Derivative):
```

Comment:
Note this will not have any effect on `Subs(Derivative(..)+1, ...)` because Derivative() + 1 is an instance of Add. Generally, this isn't the right place to insert the code that's specific to substitution in the Derivative class. That class has `_eval_subs` method for this purpose.